module/VuFind/src/VuFind/Content/Covers/LocalFile.php

```diff
@@ -0,0 +1,150 @@
+     */
+    protected $basePath;
+
+    /**
+     * Filename for LocalFile
+     *
+     * @var string
+     */
+    protected $fileName;
+
+    /**
+     * Image type for LocalFile
+     *
+     * @var string
+     */
+    protected $imageType;
+
+    /**
+     * Does this plugin support the provided ID array?
+     *
+     * @param array $ids IDs that will later be sent to load() -- see below.
+     *
+     * @return bool
+     */
+    public function supports($ids)
+    {
+        return true;
+    }
+
+    /**
+     * VuFind configuration
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Constructor
+     *
+     * @param Config $config VuFind configuration
+     */
+    public function __construct($config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * Get image location from local file storage.
+     *
+     * @param string $key  local file directory path
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        //Get settings from config.ini file
+        $config = $this->config;
+        $basePath = isset($config->$key->filepath)
+            ? $config->$key->filepath : null;
+        $fileNameToken = isset($config->$key->filename)
+            ? $config->$key->filename : null;
+        $imageType = isset($config->LocalFile->imagetype)
+            ? $config->$key->imagetype : null;
+
+        if (!isset($basePath) || !isset($fileNameToken) || !isset($imageType)) {
+            return false;
+        }
+
+        // Check base path for $VUFIND env variables
+        $vufindHome =  getenv('VUFIND_HOME');
+        $vufindLocal =  getenv('VUFIND_LOCAL_DIR') 
+        $basePath = str_replace("\$VUFIND_HOME", $vufindHome, $basePath);
+        $basePath = str_replace("\$VUFIND_LOCAL_DIR", $vufindLocal, $basePath);
+
+        // Add slash to basePath if it doesn’t exist
+        if (substr($basePath, -1) != "/") {
+            $basePath = $basePath."/";
+        }
+
+        // match fileName to information in $ids array
+        if (isset($ids[$fileNameToken])) {
+            $fileName = $ids[$fileNameToken];
+        } else {
+            return false;
+        }
+        // strip punctuation from imageType
+        $imageType = strtolower(preg_replace("/(\.)(.*)/", "$0 --> $2", $imageType));
+
+        // validate that imageType is allowed
+        $allowed_imageTypes = [ "gif", "jpg", "jpeg", "png", "tif", "tiff" ];
+        if (!in_array($imageType, $allowed_imageTypes)) {
+            throw new \Exception(
+                "Illegal file-extension '$imageType' for image '$fileName'"
+            );
+            return false;
+        }
+
+        $filepath="file:" . $basePath . $fileName . "." . $imageType;
```

Comment:
You should actually be prepending "file://" rather than just "file:" here to make this a valid file URL. For example, `file:/tmp/foo` is not legal, but `file:///tmp/foo` is legal. You might have to slightly adjust some of your processing code to match when you fix this.